.github/workflows/check-urls.yml

```diff
@@ -1,7 +1,12 @@
 name: Check URLs from changed files
 on: [push, pull_request]
+permissions:
+  contents: read
 jobs:
   job:
+    permissions:
+      # Needed for the 'trilom/file-changes-action' action
+      pull-requests: read
```

Comment:
As first step... I can say that our repository is configured by default with only read permisions for GitHub Actions.

Moreover, can you explain why and no other permissions added at each line?

And finally, `trilom/file-changes-action` is being replaced with `tj-actions/changed-files` because the number of bugs and not mantained since 2020

Reply:
It is great you have the setting set for the repository. I have no way to see it, thus my pull request. It is up to you to decide, but having it explicit in the yml may have some benefits as better portability if you ever decide to reuse it in other repositories and even narrower set of permissions.

The first change sets the global default `contents: read` permission for all jobs (currently there is only one, but I think it is a good idea to have it if at some point of time a new job is introduced). The `pull-requests: read` is added, because the `trilom/file-changes-action` is used in the job. The `actions/upload-artifact` doesn't need write permission for content. Thus, I think no other permissions are needed. Editing workflows is a bit of trial an error process because I'm not aware of a way to test it locally. However I'm committed to fix any permissions if it starts failing after being merged. Just tag me in the PR.

You are already using `- uses: trilom/file-changes-action@v1.2.4`, thus my comment. I think any possible upgrade should be done as in a separate pull request.

Reply:
Is the first time I hear about talk about this permission for trilom action. Googling a bit I found trilom/file-changes-action#129

So, you are right. LGTM :heart: